instrumentation/rack/lib/opentelemetry/instrumentation/rack/middlewares/tracer_middleware.rb

```diff
@@ -87,15 +87,11 @@ def create_frontend_span(env, extracted_context)
                                      },
                                      kind: :server)
 
-            extracted_context.set_value(current_span_key, span)
+            OpenTelemetry::Trace.context_with_span(span, parent_context: extracted_context)
           end
 
           def finish_span(context)
-            context[current_span_key]&.finish if context
-          end
-
-          def current_span_key
-            OpenTelemetry::Trace::Propagation::ContextKeys.current_span_key
+            OpenTelemetry::Trace.current_span(context)&.finish if context
```

Comment:
I don't think we need the `&.` here. `current_span` will always return non-`nil`. If it returns `Span::INVALID`, we can safely call `.finish` on it - that's a no-op on the API `Span`.